woocommerce/woocommerce-ios
# Add Initial UI for Refund Confirmation

Part of #2768. This adds the initial UI for the Refunds Summary (Confirmation) page. This is currently just using hardcoded data.  

## Design

@Garance91540 Please let me know what you think of this design implementation. 

Before | After 
--------|-------
![Simulator Screen Shot - iPhone 11 - 2020-10-01 at 16 00 09](https://user-images.githubusercontent.com/198826/94868206-f95b8300-03ff-11eb-94e4-dffaad3965d7.png)        |       ![Simulator Screen Shot - iPhone 11 (Dark) (13 7) - 2020-10-01 at 16 02 55](https://user-images.githubusercontent.com/198826/94868205-f8c2ec80-03ff-11eb-9cb9-5acf08f3eba1.png)

## Changes

I initially planned to make the [ProductSettings Mediator Pattern](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/Product%20Settings/ProductSettingsRows.swift) reusable but bailed out. I realized that it would be better to approach that when we have access to `UITableViewDiffableDataSource`. That should make things easier. 

In here, I'm following the design pattern used in [Product Editing](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/ProductFormTableViewModel.swift) but a bit lightweight and keeping things a bit reusable. I think some of these can be reused somewhere else like the rows structs:

https://github.com/woocommerce/woocommerce-ios/blob/b0e3ea981a34f6e76fde694fd0773e2e1deeaf90/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/Issue%20Refunds/RefundConfirmationViewModel.swift#L42-L67

But I'm still not sure how this is gonna pan out when the UI becomes editable. 😬 

### SettingTitleAndValueTableViewCell

Note that this cell is supposed to be reusable. It just has an outdated name. https://github.com/woocommerce/woocommerce-ios/issues/2211

### HeadlineLabelTableViewCell

I also modified this to support the “Manual Refund via Stripe” styling shown in the screenshot. 

## Testing

1. Change the contents of [this function](https://github.com/woocommerce/woocommerce-ios/blob/b0e3ea981a34f6e76fde694fd0773e2e1deeaf90/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L506-L530) to:

     ```swift
        #warning("do not commit")

        let vc = RefundConfirmationViewController()
        navigationController?.pushViewController(vc, animated: true)
        return

    ```
2. Run the app using an iOS 13.0+ device.
3. Navigate to Orders. 
4. Pick an order. 
5. The UI above should be shown. These are currently static data. 

## Submitter Checklist
